CONVENTIONS.md

```diff
@@ -29,6 +29,54 @@ Black is left at defaults, flake8 and isort are configured to adhere to black.
 
 All functions need to be covered by unit tests.
 
+# Coordinate frame conventions
+
+As the purpose of pylinalg as a library primarily is to support linear algebra
+operations in pygfx and applications based on pygfx, we adhere to a number of
+standard coordinate frame conventions to align with expectations from pygfx.
+
+Generally there are four transform steps when going from an object's local space,
+all the way to screen space:
+
+* Local space to world space: world matrix
+* World space to view space: view matrix
+* View space to NDC/clip space: projection matrix
+* NDC/clip space to screen space: screen matrix
+
+In pygfx, every world object has its own model matrix, which
+encapsulates scale, rotation and translation components. World objects
+are additionally organized hierarchically using parent/child relationships.
+The world matrix is then recursively defined for every world object; it is the
+result of multiplying an object's model matrix with its parent's world matrix.
+For the root world object, the world matrix is equal to its model matrix. 
```

Comment:
I think this is incorrect. The model matrix defines the transform w.r.t. the parent, but for the root object (the scene) there is no root. In Pygfx this works out correctly - if the camera is placed in the scene - because the camera's world matrix (i.e. the inverse of the view matrix) contains the root model matrix.

Reply:
As far as I know, we use the inverse of the camera's world matrix as the view matrix.

It's not used as the root model matrix otherwise. We have plenty of examples where the camera is not a part of the scene, and where it is part of the scene, and it's not used as the root model matrix in any instance.

Reply:
Let me try to clarify. Imagine this scene:
```
scene  --> wobjec1 --> wobject2
(root)  \--> camera1

camera2
```

Semantically, `scene.matrix` should not be part of `wobject1.world_transform` or `wobject2.world_transform`, because the matrix defines the transform w.r.t. the parent, and since the scene is the root, it's own matrix should not be taken into account. In pygfx we wrongfully do take it into account. We should fix that.

Now when this scene is rendered with camera1, we actually get correct results, because `scene.matrix` is also included in the camera's world transform, which is inverted to get the view transform.

If the scene is rendered with camera2, and `scene.matrix` is not identity, we get wrong results.

Reply:
As for this text, I think "For the root world object, the world matrix is equal to its model matrix." should either be left out or something like "For the root world object, the world matrix is the identity transform, by definition."

Reply:
> Semantically, `scene.matrix` should not be part of `wobject1.world_transform` or `wobject2.world_transform`, because the matrix defines the transform w.r.t. the parent, and since the scene is the root, it's own matrix should not be taken into account. In pygfx we wrongfully do take it into account. We should fix that.

I respectfully disagree. Counter example:

```
         /--> tire
car     /
(root)  \--> camera1

camera2
```

The car has its own transform, in order to move around world space. That world space is an empty nothingness in this example is besides the point.

> If the scene is rendered with camera2, and `scene.matrix` is not identity, we get wrong results.

Do we? I don't think so?

Reply:
>>  If the scene is rendered with camera2, and scene.matrix is not identity, we get wrong results.
>
> Do we? I don't think so?

That depends on what you expect to happen 😆

Reply:
After a discussion, we can conclude that we can choose whatever we want. As it is written now, the reference frame is the "nothingness behind the root".

Reply:
You seem to have commented earlier, but I had not seen your comment when I posted the above.

The disagreement here seems to be in the definition of the *root node* vs the *world space*. You say that they are the same. But it's equally valid to say that the world is the thing beyond the root node. It's a choice we have to make.

Personally, I initially felt that the former made more sense, but now I lean towards the latter. In part because in the case of  *root=world* we'd somehow need to enforce that the root node's transform is not used.